### PR TITLE
batched copy with offset

### PR DESCRIPTION
**Pull request by kgasperich**
changed `copy_batched` to allow incx and incy greater than 1

added a `copy_batched_offset` for easier interface

If you already have a list of pointers to the start of each array in the batch, this allows you to specify an offset that will be used for all of them instead of creating a new list of device pointers which have been shifted by the offset (we wouldn't need this if everything was uniformly sized and contiguous; we could just make a `copy_batch_strided` where we pass a single pointer each for x and y which already account for the offset)

### CODE REVIEW

File: src/Platforms/OMPTarget/ompBLAS.cpp [hunk 1]
```diff
@@ -488,12 +488,89 @@ ompBLAS_status copy_batched(ompBLAS_handle&                   handle,
...
+                            std::complex<double>* const       y[],
                             const int                         incy,
                             const int                         batch_count)
 {
   return copy_batched_impl(handle, n, x, incx, y, incy, batch_count);
 }
 #endif
+
+template<typename T>
+ompBLAS_status copy_batched_offset_impl(ompBLAS_handle& handle,
```
Comment (ye-luo) [inline src/Platforms/OMPTarget/ompBLAS.cpp:500]: BLAS APIs have offset mechanism built-in. Why is this needed?

Comment (ye-luo) [reply src/Platforms/OMPTarget/ompBLAS.cpp:500]: I mean the caller needs to add the offset to x and y arrays without the need of second API.

Comment (kgasperich) [reply src/Platforms/OMPTarget/ompBLAS.cpp:500]: could you share a link to an implementation that does that?
how would the batched equivalent of [this call](https://github.com/anbenali/qmcpack/blob/53bbc039245701bd0059e2ef01666b5f8e13273f/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp#L468) work?

I see how it would work if we are just passing a pointer to the start of the data and if we also pass a stride (assuming it is all contiguous), but if we are passing a list of pointers then I would think we'd need to make a new list where each element is incremented by the offset.

I was thinking of something where
```
for (int iw = 0; iw <nw; iw++){
  BLAS::copy(psiMinv_rows, psiMinv_temp_list_ptr[iw]+WorkingIndex, psiMinv_rows, workV2_list_ptr[iw], 1);
}
```

becomes

```
ompBLAS::copy_batched_offset(dummy_handle, psiMinv_rows, psiMinv_temp_list_ptr, WorkingIndex, psiMinv_rows, workV2_list_ptr, 0, 1, nw);
```

if we don't have the offset, then we need to make a new `psiMinv_temp_list_ptr_shifted` where `psiMinv_temp_list_ptr_shifted[iw] = psiMinv_temp_list_ptr[iw] + WorkingIndex`

Comment (ye-luo) [reply src/Platforms/OMPTarget/ompBLAS.cpp:500]: yes. Build a new list with each element increase by the offset.

Comment (kgasperich) [reply src/Platforms/OMPTarget/ompBLAS.cpp:500]: yes, it was mostly just for convenience to avoid the need to build the new list
